# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/power/WakeLockSentinel.cpp
# Commit: b55991835aec
# Full Hash: b55991835aecc36d7b8cf86d753c54847bd8b0ec
# Author: Vincent Hilla <vhilla@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1589554
# File Overlap Count: 1
# Description:
#   Bug 1589554 - Part 1: Screen Wake Lock boilerplate. r=dom-core,smaug,emilio,edgar
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189508
# ==============================================================================

diff -r 8c57ecb4b95f -r b55991835aec dom/power/WakeLockSentinel.cpp
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/dom/power/WakeLockSentinel.cpp	Tue Dec 05 15:37:05 2023 +0000
@@ -0,0 +1,21 @@
+/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim: set ts=8 sts=2 et sw=2 tw=80: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/dom/WakeLockSentinelBinding.h"
+#include "WakeLockSentinel.h"
+
+namespace mozilla::dom {
+
+JSObject* WakeLockSentinel::WrapObject(JSContext* cx,
+                                       JS::Handle<JSObject*> aGivenProto) {
+  return WakeLockSentinel_Binding::Wrap(cx, this, aGivenProto);
+}
+
+bool WakeLockSentinel::Released() const { return false; }
+
+already_AddRefed<Promise> WakeLockSentinel::ReleaseLock() { return nullptr; }
+
+}  // namespace mozilla::dom